File: examples/kotlin/src/test/kotlin/OpenBrowserTest.kt
import org.junit.jupiter.api.Disabled
import org.junit.jupiter.api.Test
import org.openqa.selenium.WebDriver
import org.openqa.selenium.chrome.ChromeDriver
import org.openqa.selenium.chrome.ChromeOptions
import org.openqa.selenium.edge.EdgeDriver
import org.openqa.selenium.edge.EdgeOptions
import org.openqa.selenium.firefox.FirefoxDriver
import org.openqa.selenium.firefox.FirefoxOptions
import org.openqa.selenium.ie.InternetExplorerDriver
import org.openqa.selenium.ie.InternetExplorerOptions
import org.openqa.selenium.safari.SafariDriver
import org.openqa.selenium.safari.SafariOptions


class OpenBrowserTest {
    private var driver: WebDriver? = null

    @Test
    fun chromeSession() {
        val options = ChromeOptions()
        driver = ChromeDriver(options)
        (driver as ChromeDriver).quit()
    }

    @Test
    fun edgeSession() {
        val options = EdgeOptions()
        driver = EdgeDriver(options)
        (driver as EdgeDriver).quit()
    }

    @Test
    fun firefoxSession() {
        val options = FirefoxOptions()
        driver = FirefoxDriver(options)
        (driver as FirefoxDriver).quit()
    }

    @Disabled("Only runs on Windows")
    @Test
    fun internetExplorerSession() {
        val options = InternetExplorerOptions()
        driver = InternetExplorerDriver(options)
        (driver as InternetExplorerDriver).quit()
    }

    @Disabled("Only runs on Windows")
    @Test
    fun internetExplorerCompatibilitySession() {
        val options = InternetExplorerOptions()
        options.attachToEdgeChrome()
        options.withEdgeExecutablePath("/path/to/edge/browser")
        driver = InternetExplorerDriver(options)
        (driver as InternetExplorerDriver).quit()
    }

    @Disabled("Requires non-standard browser")
    @Test
    fun operaSession() {
        val options = ChromeOptions()
        options.setBinary("/path/to/opera/browser")
        driver = ChromeDriver(options)
        (driver as ChromeDriver).quit()
    }

    @Disabled("Only runs on Mac")
    @Test
    fun safariSession() {
        val options = SafariOptions()
        driver = SafariDriver(options)
        driver!!.quit()
    }
}

File: examples/scala/src/test/scala/OpenBrowserTest.scala
import org.openqa.selenium.chrome.{ChromeDriver, ChromeOptions}
import org.openqa.selenium.edge.{EdgeDriver, EdgeOptions}
import org.openqa.selenium.firefox.{FirefoxDriver, FirefoxOptions}
import org.openqa.selenium.ie.{InternetExplorerDriver, InternetExplorerOptions}
import org.openqa.selenium.safari.{SafariDriver, SafariOptions}
import org.scalatest.BeforeAndAfterAll
import org.scalatest.concurrent.Eventually
import org.scalatest.flatspec.AnyFlatSpec
import org.scalatest.matchers.should.Matchers
import org.scalatestplus.selenium.WebBrowser

class OpenBrowserTest extends AnyFlatSpec with BeforeAndAfterAll with Matchers with Eventually with WebBrowser {

  it should " be able to launch chrome" in {
    val chromeOptions = new ChromeOptions()
    val chromeDriver = new ChromeDriver(chromeOptions)
    chromeDriver.quit()
  }

  it should " be able to launch firefox" in {
    val firefoxOptions = new FirefoxOptions()
    val firefoxDriver = new FirefoxDriver(firefoxOptions)
    firefoxDriver.quit()
  }

  it should " be able to launch edge" in {
    val edgeOptions = new EdgeOptions()
    val edgeDriver = new EdgeDriver(edgeOptions)
    edgeDriver.quit()
  }

  ignore should " be able to launch ie" in {
    val ieOptions = new InternetExplorerOptions()
    ieOptions.attachToEdgeChrome()
    ieOptions.withEdgeExecutablePath("/path/to/ie/browser")
    val ieDriver = new InternetExplorerDriver(ieOptions)
    ieDriver.quit()
  }

  ignore should " be able to launch opera" in {
    val chromeOptions = new ChromeOptions()
    chromeOptions.setBinary("/path/to/opera/browser")
    val ieDriver = new ChromeDriver(chromeOptions)
    ieDriver.quit()
  }

  ignore should " be able to launch safari" in {
    val safariOptions = new SafariOptions()
    val safariDriver = new SafariDriver(safariOptions)
    safariDriver.quit()
  }

}
